examples/custom_game_data/main.rs

```diff
@@ -111,54 +111,62 @@ impl<'a, 'b> State<CustomGameData<'a, 'b>> for Loading {
     }
 }
 
-impl<'a, 'b> State<CustomGameData<'a, 'b>> for Paused {
+impl<'a, 'b> State<CustomGameData<'a, 'b>, ()> for Paused {
     fn handle_event(
         &mut self,
         data: StateData<CustomGameData>,
-        event: Event,
-    ) -> Trans<CustomGameData<'a, 'b>> {
-        if is_close_requested(&event) || is_key_down(&event, VirtualKeyCode::Escape) {
-            Trans::Quit
-        } else if is_key_down(&event, VirtualKeyCode::Space) {
-            let _ = data.world.delete_entity(self.ui);
-            Trans::Pop
+        event: StateEvent<()>,
+    ) -> Trans<CustomGameData<'a, 'b>, ()> {
+        if let StateEvent::Window(event) = &event {
+            if is_close_requested(&event) || is_key_down(&event, VirtualKeyCode::Escape) {
```

Comment:
The thing with is_close_requested and is_key_down is that they can be called internally from the engine. The engine crates cannot know about the `StateEvent` type because of circular references, so it is not possible at all to change both of those to use StateEvent.

Reply:
Okay, thanks for clarifying!